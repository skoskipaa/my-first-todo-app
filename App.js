import React, { useState } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import ToDoList from './components/ToDoList'
import ToDoInput from './components/ToDoInput'

export default function App() {
  const [toDos, setToDos] = useState([])
  const [isVisible, setVisible] = useState(false)

  const addToDo = (newToDo) => {
    setToDos(currentState => [...currentState, 
      {
        id: Math.random().toString(),
        content: newToDo
      }])
    handleVisibility()
  }

  const deleteToDo = (id) => {
    setToDos(currentToDos => {
      return toDos.filter(todo => todo.id !== id)
    })
  }
  //setToDos(toDos.filter(todo => todo.id !== id))

  const handleVisibility = () => {
    setVisible(!isVisible)
  }

  return (
    <View style={styles.container}>
      <Text style={styles.appTitle}>My Awesome ToDo App</Text>
      <Button title="Add a new ToDo" onPress={handleVisibility}/>
      <ToDoInput addToDo={addToDo} isVisible={isVisible} handleVisibility={handleVisibility}/>
      <ToDoList toDos={toDos} handleDelete={deleteToDo}/>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f9ebff',
    alignItems: 'center',
    paddingTop: 30,
  },
  appTitle: {
    color: '#361470',
    fontSize: 24,
  },
});
