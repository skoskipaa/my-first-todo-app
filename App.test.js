import React from 'react'
import renderer from 'react-test-renderer'

import App from './App'

describe('The <App /> component', () => {
    it('has 4 children', () => {
        const tree = renderer.create(<App />).toJSON()
        expect(tree.children.length).toBe(4)
    })

    // use 'yarn test --updateSnapshot' after UI changes
    it('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
