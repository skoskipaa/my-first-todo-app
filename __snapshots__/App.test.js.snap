// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The <App /> component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f9ebff",
      "flex": 1,
      "justifyContent": "center",
      "paddingTop": 30,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#361470",
        "fontSize": 24,
      }
    }
  >
    My Awesome ToDo App
  </Text>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add a new ToDo
      </Text>
    </View>
  </View>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        rejectResponderTermination={true}
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "fontSize": 18,
            "height": 40,
            "marginBottom": 15,
            "padding": 3,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "width": "60%",
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add
            </Text>
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "red",
                  },
                ]
              }
            >
              Cancel
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
